service/dap/server_test.go

```diff
@@ -442,7 +536,472 @@ func TestStackTraceRequest(t *testing.T) {
+			}, {
+				// Stop at line 25
+				execute: func() {
+					// Frame ids get reset at each breakpoint.
+					client.StackTraceRequest(1, 0, 20)
+					stack := client.ExpectStackTraceResponse(t)
+					expectStackFrames(t, stack, 25, 1000, 5, 5)
+
+					client.ScopesRequest(1000)
+					scopes := client.ExpectScopesResponse(t)
+					expectScope(t, scopes, 0, "Arguments", 1000)
+					expectScope(t, scopes, 1, "Locals", 1001)
+
+					client.ScopesRequest(1111)
+					erres := client.ExpectErrorResponse(t)
+					if erres.Body.Error.Format != "Unable to list locals: unknown frame id 1111" {
+						t.Errorf("\ngot %#v\nwant Format=\"Unable to list locals: unknown frame id 1111\"", erres)
+					}
+
+					client.VariablesRequest(1000) // Arguments
+					args := client.ExpectVariablesResponse(t)
+					expectChildren(t, args, "Arguments", 0)
+
+					client.VariablesRequest(1001) // Locals
+					locals := client.ExpectVariablesResponse(t)
+					expectChildren(t, locals, "Locals", 1)
+					expectVarExact(t, locals, -1, "a1", `"bur"`, noChildren)
+
+					client.VariablesRequest(7777)
+					erres = client.ExpectErrorResponse(t)
+					if erres.Body.Error.Format != "Unable to lookup variable: unknown reference 7777" {
+						t.Errorf("\ngot %#v\nwant Format=\"Unable to lookup variable: unknown reference 7777\"", erres)
+					}
+				},
+				disconnect: false,
+			}})
+	})
+}
+
+// TestScopesAndVariablesRequests2 executes to a breakpoint and tests different
+// configurations of 'scopes' and 'variables' requests.
+func TestScopesAndVariablesRequests2(t *testing.T) {
+	runTest(t, "testvariables2", func(client *daptest.Client, fixture protest.Fixture) {
+		runDebugSessionWithBPs(t, client,
+			// Launch
+			func() {
+				client.LaunchRequest("exec", fixture.Path, !stopOnEntry)
+			},
+			// Breakpoints are set within the program
+			fixture.Source, []int{},
+			[]onBreakpoint{{
+				// Stop at line 317
+				execute: func() {
+					client.StackTraceRequest(1, 0, 20)
+					stack := client.ExpectStackTraceResponse(t)
+					expectStackFrames(t, stack, 317, 1000, 3, 3)
+
+					client.ScopesRequest(1000)
+					scopes := client.ExpectScopesResponse(t)
+					expectScope(t, scopes, 0, "Arguments", 1000)
+					expectScope(t, scopes, 1, "Locals", 1001)
+				},
+				disconnect: false,
+			}, {
+				// Stop at line 322
+				execute: func() {
+					client.StackTraceRequest(1, 0, 20)
+					stack := client.ExpectStackTraceResponse(t)
+					expectStackFrames(t, stack, 322, 1000, 3, 3)
+
+					client.ScopesRequest(1000)
+					scopes := client.ExpectScopesResponse(t)
+					expectScope(t, scopes, 0, "Arguments", 1000)
+					expectScope(t, scopes, 1, "Locals", 1001)
+
+					// Arguments
+
+					client.VariablesRequest(1000)
+					args := client.ExpectVariablesResponse(t)
+					expectChildren(t, args, "Arguments", 0)
+
+					// Locals
+
+					client.VariablesRequest(1001)
+					locals := client.ExpectVariablesResponse(t)
+					expectChildren(t, locals, "Locals", 93)
```

Comment:
Given the problem with Go 1.15 I'd say we shouldn't test this. Besides you are already checking the variables individually, this shouldn't break if we add a new one.

Reply:
I already reworked the test, so 1.15 changes don't cause different behavior for this check.

Reply:
I see, still I don't think we should do this test, it doesn't add much and it'll break if we add something to testvariables2.go.

Reply:
Done